node/custom.go

```diff
@@ -0,0 +1,39 @@
+package node
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/INFURA/go-ethlibs/jsonrpc"
+)
+
+type customTransport struct {
+	requester  Requester
+	subscriber Subscriber
+}
+
+func (t *customTransport) Request(ctx context.Context, r *jsonrpc.Request) (*jsonrpc.RawResponse, error) {
+	return t.requester.Request(ctx, r)
+}
+
+func (t *customTransport) Subscribe(ctx context.Context, r *jsonrpc.Request) (Subscription, error) {
+	if t.subscriber == nil {
+		return nil, errors.New("subscriptions not supported over this transport")
+	}
+
+	return t.subscriber.Subscribe(ctx, r)
+}
+
+func (t *customTransport) IsBidirectional() bool {
+	return t.subscriber != nil
+}
+
+func newCustomTransport(requester Requester, subscriber Subscriber) (*customTransport, error) {
```

Comment:
Optional (thinking aloud): Seems so trivial, could make this a public struct and skip the constructor altogether.

Reply:
Or actually, it wouldn't even need to be public, but could just skip the constructor until it does something meaningful. /shrug

Reply:
Ya, I think with a later refactor I'd like to make Transport itself public and make it _have_ a Requester and Subscriber instead of _be_ one.  But I'm going to save that refactor until I'm not under the gun on other projects. :) (so who knows when.. ;) )